be/src/exec/tablet_scanner.cpp

```diff
@@ -137,8 +137,16 @@ Status TabletScanner::_init_reader_params(const std::vector<OlapScanRange*>* key
     // we will not call agg object finalize method in scan node,
     // to avoid the unnecessary SerDe and improve query performance
     _params.need_agg_finalize = _need_agg_finalize;
-    _params.use_page_cache = !config::disable_storage_page_cache;
 
+    const TQueryOptions& query_options = _runtime_state->query_options();
+    if (query_options.__isset.use_page_cache) {
+        _params.use_page_cache = query_options.use_page_cache;
+        if (_params.use_page_cache && config::disable_storage_page_cache) {
+            _params.use_page_cache = false;
+        }
+    } else {
+        _params.use_page_cache = !config::disable_storage_page_cache;
+    }
```

Comment:
TQueryOptions is not required to leaked to TabletScanner, we can put this block of code into  RuntimeState::use_page_cache method. it seems use_page_cache = true is a default behavior.

```
bool RuntimeState::use_page_cache(){
    const TQueryOptions& query_options = _runtime_state->query_options();
    if (query_options.__isset.use_page_cache) {
        _params.use_page_cache = query_options.use_page_cache;
        if (!_params.use_page_cache || config::disable_storage_page_cache) {
             return false;
        }
    } else {
        return !config::disable_storage_page_cache;
    }
}
```

Reply:
> TQueryOptions is not required to leaked to TabletScanner, we can put this block of code into RuntimeState::use_page_cache method. it seems use_page_cache = true is a default behavior.
> 
> ```
> bool RuntimeState::use_page_cache(){
>     const TQueryOptions& query_options = _runtime_state->query_options();
>     if (query_options.__isset.use_page_cache) {
>         _params.use_page_cache = query_options.use_page_cache;
>         if (!_params.use_page_cache || config::disable_storage_page_cache) {
>              return false;
>         }
>     } else {
>         return !config::disable_storage_page_cache;
>     }
> }
> ```

updated